server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailConverter.java

```diff
@@ -84,10 +84,19 @@ DeletedMessage fromMail(Mail mail) {
             .sender(mail.getMaybeSender())
             .recipients(mail.getRecipients())
             .hasAttachment(retrieveHasAttachment(mail))
+            .bodySize(getMessageSize(mail))
             .subject(retrieveSubject(mail))
             .build();
     }
 
+    private long getMessageSize(Mail mail) {
+        try {
+            return mail.getMessageSize();
+        } catch (MessagingException e) {
+            throw new RuntimeException("can not get message size from message " + mail.getName(), e);
```

Comment:
Not sure propagating the failure will be the best idea ever here.

How about  simply counting bytes in theMimeMessage as a fallback?

`return MimeMessageUtil.asBytes(mail.getMessage()).length;`

(And if it fails again then lets propagate it....)